packages/kit/src/core/sync/write_types.js

```diff
@@ -449,36 +449,43 @@ function process_node(ts, node, outdir, params, groups) {
 		server_data = 'null';
 	}
 
+	const parent_type = get_parent_type('LayoutData');
+
 	if (node.shared) {
 		const content = fs.readFileSync(node.shared, 'utf8');
 		const proxy = tweak_types(ts, content, shared_names);
 		if (proxy?.modified) {
 			written_proxies.push(write(`${outdir}/proxy${path.basename(node.shared)}`, proxy.code));
 		}
 
-		data = get_data_type(node.shared, 'load', server_data, proxy);
-		load = `Kit.Load<${params}, ${server_data}, ${get_parent_type('LayoutData')}, OutputData>`;
+		data = `${parent_type} & ${get_data_type(
+			node.shared,
+			`${parent_type} & ${server_data}`,
+			proxy
+		)}`;
+		load = `Kit.Load<${params}, ${server_data}, ${parent_type}, OutputData>`;
+	} else if (server_data === 'null') {
+		data = parent_type;
 	} else {
-		data = server_data;
+		data = `${parent_type} & ${server_data}`;
```

Comment:
This could give a type conflict if they set the same prop but to different types. It probably needs the same omit logic as the parent type function.

Reply:
good catch, added